.circleci/setup_env.sh

```diff
@@ -54,11 +54,7 @@ if pip list | grep -q ^pandas; then
     pip uninstall -y pandas || true
 fi
 
-echo "Build extensions"
-# GH 47305: Parallel build can causes flaky ImportError from pandas/_libs/tslibs
-python setup.py build_ext -q -j1
-
 echo "Install pandas"
-python -m pip install --no-build-isolation --no-use-pep517 -e .
+python -m pip install --no-build-isolation -v .
```

Comment:
Do we still need build isolation?

Reply:
We'll turn it back on eventually. Right now, I don't think you can specify git dependencies in pyproject.toml.

We'll probably need to wait for meson 0.65.0(needs mesonbuild/meson#11000 and a fix for detecting MSVC when it's not in path in.) and a new version of meson-python. 

I'm not too worried about meson-python, though since we can basically cut a release there anytime.

Reply:
> Right now, I don't think you can specify git dependencies in pyproject.toml.

meson master should be equivalent to 0.64.0 I'd think, that release happened last week.

> needs https://github.com/mesonbuild/meson/pull/11000

I'll see if I can test that with SciPy later today. I have to undo all the workaround we added for that to reproduce the issue.